emily/handler/src/api/models/withdrawal/responses.rs

```diff
@@ -24,11 +24,17 @@ pub struct UpdateWithdrawalsResponse {
 }
 
 /// Wrapper for withdrawal with status code. Used for multi-status responses.
+/// TODO: utopia, which we use for generating OpenAPI spec does not support
+/// [`Result`] in structs, however, logically exactly one of `withdrawal` or `error` should be
+/// None, and exactly one of them should be Some. It would be nice to find a way to use
+/// `Result` here.
```

Comment:
I think the returned JSON cannot really support returning a result, since it would need to be an object, and then you arrive at something similar to what we have here. So I would just note that, logically, exactly one of `withdrawal` or `error` will be `None` and I'd remove the TODO since there is nothing TODO.

If we want to enforce this only-one-will-be-none invariant in code, which I think we want to do, we can make it so that the only way to construct this type would be through a `WithdrawalWithStatus::from_result` function that takes a `Result` and maybe a status code, and properly creates the struct. This way the struct enforces its invariants, and we cannot mess it up. Same thing for the `DepositWithStatus` struct.

Reply:
Well, on json level it really sounds like impossible to have a result. But maybe it can be done somehow on rust level in such a way that during json parsing an error will be raised if this json can not be parsed into valid `Result`

Reply:
[0ba63ef](https://github.com/stacks-sbtc/sbtc/pull/1743/commits/0ba63ef59ca21e2f84d912823c3166b4a836e033)